src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertyAccessorSymbol.cs

```diff
@@ -410,7 +410,12 @@ private DeclarationModifiers MakeModifiers(AccessorDeclarationSyntax syntax, Loc
             const DeclarationModifiers defaultAccess = DeclarationModifiers.None;
 
             // Check that the set of modifiers is allowed
-            const DeclarationModifiers allowedModifiers = DeclarationModifiers.AccessibilityMask;
+            DeclarationModifiers allowedModifiers = DeclarationModifiers.AccessibilityMask;
```

Comment:
Do we need to check that the parent property this accessor belongs to isn't readonly? 

For access modifiers it's an error when you specify an accessor modifier that isn't more restrictive https://sharplab.io/#v2:D4AQDABCCMDcCwAoEBmKAmCBhCBvJEhBhUaMkAysYfoiYQJADmApgC4J30QOoQDO7TvQC+SEUA==

Reply:
It seems like that would be consistent with the behavior for visibility modifiers.

Do you propose giving a similar error for this case?
```cs
struct S
{
    readonly string P
    {
     	readonly get;
    }
}
```

Reply:
Yeah, that's what I was thinking. Should we be giving an error in that case? I don't really have an opinion one way or the other, but it seems like it would be consistent if we did.

---
In reply to: [260932737](https://github.com/dotnet/roslyn/pull/32888#discussion_r260932737) [](ancestors = 260932737)

Reply:
I went ahead and made it an error to provide readonly on an accessor for a property marked readonly.